base/broadcast.jl

```diff
@@ -431,57 +510,177 @@ julia> (1 + im) ./ Nullable{Int}()
 Nullable{Complex{Float64}}()
 ```
 """
-@inline broadcast(f, A, Bs...) = broadcast_c(f, containertype(A, Bs...), A, Bs...)
+@inline broadcast(f, A, Bs...) =
+    broadcast(f, Result{combine_types(A, Bs...)}(), A, Bs...)
+
+"""
+    broadcast(f, Broadcast.Result{ContainerType}(), As...)
+
+Specify the container-type of the output of a broadcasting operation.
+You can specialize such calls as
+
+    function Broadcast.broadcast(f, ::Broadcast.Result{ContainerType,Void,Void}, As...) where ContainerType
+        ...
+    end
+"""
+@inline function broadcast(f, ::Result{ContainerType,Void,Void}, A, Bs...) where ContainerType
+    ElType = combine_eltypes(f, A, Bs...)
+    broadcast(f,
+              Result{ContainerType,ElType}(combine_indices(A, Bs...)),
+              A, Bs...)
+end
+
+"""
+    broadcast(f, Broadcast.Result{ContainerType,ElType}(indices), As...)
+
+Specify the container-type, element-type, and indices of the output
+of a broadcasting operation. You can specialize such calls as
+
+    function Broadcast.broadcast(f, r::Broadcast.Result{ContainerType,ElType,<:Tuple}, As...) where {ContainerType,ElType}
+        ...
+    end
+
+This variant might be the most convenient specialization for container types
+that don't support [`setindex!`](@ref) and therefore can't use [`broadcast!`](@ref).
```

Comment:
Likewise here, pointing to the relevant manual section for more might be worthwhile?